File: p26/p26.erl
#!/usr/bin/env escript
%% -*- erlang -*-
%%! -smp enable -sname p26
% vim:syn=erlang

-mode(compile).


main(_) ->
    {_,Answer} = lists:last(lists:sort([ test26(X, []) || X <- lists:seq(1,10000)])),
    io:format("Answer: ~p ~n", [Answer]).

test26(N, [])      -> test26(N, [inc(1,N) rem N]);
test26(_, [0|_])   -> 0;
test26(N, [H|_]=L) ->
    D = inc(H,N) rem N,
    case lists:member(D, L) of
        true  -> {length(L) - indx(D, lists:reverse(L), 0), N}
        ;_ -> test26(N, [D | L])
    end.

indx(D, [D|_], I) -> I;
indx(D, [_|T], I) -> indx(D, T, I+1).

inc(H,N) when H >= N -> H;
inc(H,N) -> inc(H*10, N).

File: p26/p26.py
#!/usr/bin/python3

def recurring_cycle(n, d):
    for t in range(1, d):
        if 1 == pow(10, t, d):
            return (t,d)
    return (0,d)

Answer = sorted([recurring_cycle(1, i) for i in range(2,1001)], key = lambda tup: tup[0])[-1][1]

print("Answer %s" % Answer)
